### PR TITLE
Add a round trip test for persist package.

### PR DESCRIPTION
**Pull request by notbdu**
Adds a round trip test for writing data to a file and reading fields from disk.

### CODE REVIEW

**Comment by notbdu** [comment]
Also fixes bug: #144.

File: persist/fs/equals_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	iter2, err := f2.Values().Iter()
+	if err != nil {
+		return false, err
+	}
+	for iter1.Next() && iter2.Next() {
+		if iter1.Current() != iter2.Current() {
+			return false, nil
+		}
+	}
+	// If either iterator has extra data, they are not equal.
```
Comment (xichen2020) [inline persist/fs/equals_test.go]: Need to also check `iter1.Err()` and `iter2.Err()` are nil? Same for other functions in the file.

File: persist/fs/reader.go [hunk 2]
```diff
@@ -390,11 +390,12 @@ func (r *reader) mmapReadAllAndValidateChecksum(fd *os.File) ([]byte, error) {
...
 		r.logger.Warnf("warning during memory mapping info file %s: %s", fd.Name(), res.Warning)
 	}
-	if err := digest.Validate(res.Result); err != nil {
+	validatedBytes, err := digest.Validate(res.Result)
+	if err != nil {
 		mmap.Munmap(res.Result)
 		return nil, err
 	}
-	return res.Result, nil
+	return validatedBytes, nil
```
Comment (xichen2020) [inline persist/fs/reader.go]: I think you should return a `cleanup` function to `munmap(res.Result)` in addition to `validatedBytes`, as that's only a portion of the mmap'ed region.

**Comment by xichen2020** [comment]
Nice find!

File: persist/fs/equals_test.go (hunk 1 above)
Comment (xichen2020) [reply persist/fs/equals_test.go]: You can just do `require.NoError(t, iter1.Err())` for example.

**Pull request merged**